src/transform/parameters.ts

```diff
@@ -39,9 +41,13 @@ export function transformParametersArray(
       if (paramComment) output += comment(paramComment);
 
       const required = paramObj.required ? `` : `?`;
-      output += `    "${paramName}"${required}: ${
-        paramObj.schema ? transformSchemaObj(paramObj.schema) : "unknown"
-      };\n`;
+      let type = ``;
+      if (version === 2) {
+        type = paramObj.type ? transformSchemaObj(paramObj) : "unknown";
```

Comment:
So according to the [OpenAPI 2.0 spec](https://swagger.io/specification/v2/#parameterObject), this will cause a regression, which is why the V2 tests are failing (to your credit, I think the specs are **very** confusing here, and I suspect it’s why they changed it for 3.0 🙂 ).

Instead, this needs to be:

```suggestion
  if (paramObj.in === 'body' && paramObj.schema) {
    type = transformSchemaObj(paramObj.schema);
  } else if (paramObj.type) {
    type = transformSchemaObj(paramObj);
  } else {
    type = 'unknown';
  }
```

Reply:
Thank you! I missed the spec.. fixed in https://github.com/drwpow/openapi-typescript/pull/489/commits/fe3b69b4a0682f9799ffef29ce57dcc498cc47f4